### ISSUE TITLE
EvalResult doesn't do mean_of_gpus if using TensorMetric

### ISSUE THREAD

**Issue by xiadingZ** [PyTorchLightning/pytorch-lightning#2795]
I want to use new `AccuracyMetric`, it can automatically sync in ddp, but it doesn't divide by `word_size`. In manually mode, I can  divide it by `word_size` by hand in `validation_epoch_end`. But if I use `EvalResult`, how to do this? It only do `mean`  across batches, but no across gpus.
 This is original code:
```
    def validation_epoch_end(self, outputs):
        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
        avg_acc = torch.stack([x['acc'] for x in outputs]).mean()
        avg_acc = avg_acc / self.trainer.world_size

```

**Comment by Borda** [comment]
@SkafteNicki @justusschock mind have a look?

**Comment by SkafteNicki** [comment]
@xiadingZ this was solved in a recent PR (https://github.com/PyTorchLightning/pytorch-lightning/pull/2568). You can now set the `reduce_op='avg'` when you construct the metric, and it will calculate the mean and not the sum.
